### DISCUSSION TITLE
hydra with libssh not working M1 Max, Monterey

### DISCUSSION THREAD

**Discussion by benasvaleika** [Homebrew/discussions#2781] [Everyday usage]
Hello, i can't get hydra working for libssh on mac M1 Max, Monterey. 

brew info hydra returns: 

```
hydra: stable 9.2 (bottled), HEAD
Network logon cracker which supports many services
https://github.com/vanhauser-thc/thc-hydra
Conflicts with:
  ory-hydra (because both install `hydra` binaries)
Not installed
From: https://github.com/Homebrew/homebrew-core/blob/HEAD/Formula/hydra.rb
License: AGPL-3.0-only
==> Dependencies
Build: pkg-config ✔
Required: libssh ✔, mysql-client ✔, openssl@1.1 ✔
==> Options
--HEAD
	Install HEAD version
==> Analytics
install: 724 (30 days), 2,647 (90 days), 14,352 (365 days)
install-on-request: 723 (30 days), 2,639 (90 days), 14,212 (365 days)
build-error: 0 (30 days)
```

However, when i run hydra with ssh configuration, 
"[ERROR] Compiled without LIBSSH v0.4.x support, module is not available!"
is returned.

I've tried manually uninstalling/installing latest stable libssh, did not solve the problem.

Any help appreciated.

**Comment by gromgit** [comment]
>```
> hydra: stable 9.2 (bottled), HEAD
> Network logon cracker which supports many services
> https://github.com/vanhauser-thc/thc-hydra
> Conflicts with:
> ory-hydra (because both install hydra binaries)
> Not installed
>```
Note that last line. Based on your `brew info` output, the `hydra` you're running isn't a Homebrew one.

**Reply by benasvaleika** [reply]
My bad, for running brew info while hydra was uninstalled. (was also messing with hydra's source binary)

Below I paste the same command with Homebrew hydra installed, as well as attempt to use ssh with hydra (ssh://x.x.x.x)
Note, that all other hydra's options that do not require libssh, run just fine.

## Brew info
```
hydra: stable 9.2 (bottled), HEAD
Network logon cracker which supports many services
https://github.com/vanhauser-thc/thc-hydra
Conflicts with:
  ory-hydra (because both install `hydra` binaries)
/opt/homebrew/Cellar/hydra/9.2 (16 files, 1.5MB) *
  Poured from bottle on 2022-01-14 at 12:55:55
From: https://github.com/Homebrew/homebrew-core/blob/HEAD/Formula/hydra.rb
License: AGPL-3.0-only
==> Dependencies
Build: pkg-config ✔
Required: libssh ✔, mysql-client ✔, openssl@1.1 ✔
==> Options
--HEAD
	Install HEAD version
==> Analytics
install: 727 (30 days), 2,643 (90 days), 14,350 (365 days)
install-on-request: 725 (30 days), 2,634 (90 days), 14,209 (365 days)
build-error: 0 (30 days)
```
## hydra with ssh://x.x.x.x
```
Hydra v9.2 (c) 2021 by van Hauser/THC & David Maciejak - Please do not use in military or secret service organizations, or for illegal purposes (this is non-binding, these *** ignore laws and ethics anyway).

Hydra (https://github.com/vanhauser-thc/thc-hydra) starting at 2022-01-14 12:57:19
[WARNING] Many SSH configurations limit the number of parallel tasks, it is recommended to reduce the tasks: use -t 4
[ERROR] Compiled without LIBSSH v0.4.x support, module is not available!
```

It seems that hydra does not see libssh during compilation.

Previously it was possible to fix this exact problem by running
`brew install hydra --with-libssh`
(src: https://itectec.com/superuser/how-to-get-thc-hydra-to-work-from-homebrew-installation/)
However, it seems like this option no longer exists.

**Reply by carlocab** [reply]
What is the output of `brew linkage hydra`? It seems that `hydra` may not have linked with `libssh` properly during build time.

**Reply by benasvaleika** [reply]
Correct, linkage between hydra and libssh was not successful.
```
Dependencies with no linkage:
  libssh
```

**Reply by benasvaleika** [reply]
I also add brew info libssh:
```
brew info libssh
libssh: stable 0.9.6 (bottled), HEAD
C library SSHv1/SSHv2 client and server protocols
https://www.libssh.org/
/opt/homebrew/Cellar/libssh/0.9.6 (23 files, 1.2MB) *
  Poured from bottle on 2022-01-14 at 00:36:40
From: https://github.com/Homebrew/homebrew-core/blob/HEAD/Formula/libssh.rb
License: LGPL-2.1-or-later
==> Dependencies
Build: cmake ✔
Required: openssl@1.1 ✔
==> Options
--HEAD
	Install HEAD version
==> Analytics
install: 24,638 (30 days), 81,362 (90 days), 293,671 (365 days)
install-on-request: 1,579 (30 days), 14,685 (90 days), 79,609 (365 days)
build-error: 0 (30 days)
```

**Reply by carlocab** [reply]
The formula needs to be adjusted to fix that. Do you want to take a swing at it? Otherwise, I can try to take a look, but that might not be for a while.

**Reply by benasvaleika** [reply]
I'll take a look at the formula, however, because my knowledge of ruby scripting is quite limited, I doubt I'll be able to solve it on my own. Would appreciate someone else taking a look at it.

**Reply by gromgit** [reply]
Opened https://github.com/Homebrew/homebrew-core/pull/93150 to fix this.
